app/src/features/projects/view/components/FundingSource.tsx

```diff
@@ -19,24 +30,111 @@ const useStyles = makeStyles({
   }
 });
 
-export interface IProjectDetailsProps {
+export interface IProjectFundingProps {
   projectForViewData: IGetProjectForViewResponse;
+  codes: IGetAllCodeSetsResponse;
+  refresh: () => void;
 }
 
 /**
  * Funding source content for a project.
  *
  * @return {*}
  */
-const FundingSource: React.FC<IProjectDetailsProps> = (props) => {
+const FundingSource: React.FC<IProjectFundingProps> = (props) => {
   const {
-    projectForViewData: { funding }
+    projectForViewData: { funding, id },
+    codes: codes
   } = props;
 
   const classes = useStyles();
+  const biohubApi = useBiohubApi();
+
+  const [errorDialogProps, setErrorDialogProps] = useState<IErrorDialogProps>({
+    dialogTitle: EditFundingI18N.editErrorTitle,
+    dialogText: EditFundingI18N.editErrorText,
+    open: false,
+    onClose: () => {
+      setErrorDialogProps({ ...errorDialogProps, open: false });
```

Comment:
can we add a test case for when the error dialog gets closed? similar to how other components do it